question_answering/question_answering.py

```diff
@@ -50,158 +405,284 @@ def _get_prompt_template(
-                if len(answers) != len(questions):
-                    continue
-                answers = [
-                    answer[3:] for answer in answers
-                ]  # Without the questions index prefix 'i. '
-                if all(len(answer) == 0 for answer in answers):
-                    continue
-                # Note in the dataframe:
-                df.loc[i - len(errors)] = [
-                    str(text_file.relative_to(text_files_directory)),
-                    *answers,
+            # Take i's answer:
+            else:
+                answer_i, text = text.split(f"{i + 1}.", 1)
+            # Collect the answer removing redundant spaces:
+            answers.append(answer_i.strip())
+
+        return answers
+
+    def _answer_questions(
+        self,
+        questions_amount: int,
+        batched_input: List[str],
+        generation_pipeline: transformers.Pipeline,
+        generation_config: transformers.GenerationConfig,
+    ) -> List[List[str]]:
+        # Infer through the llm:
+        batched_output = generation_pipeline(
+            batched_input,
+            generation_config=generation_config,
+            eos_token_id=generation_pipeline.tokenizer.eos_token_id,
+            return_full_text=False,
+            num_return_sequences=1,
+        )
+        # Process the outputs to get the answers:
+        batched_answers = []
+        for output in batched_output:
+            # Get the generated answers:
+            answers = self._get_answers(
+                generated_text=output[0]["generated_text"],
+                questions_amount=questions_amount,
+            )
+            # Collect the processed answers:
+            batched_answers.append(answers)
+        return batched_answers
+
+    def handle_questions(
+        self,
+        questions_amount: int,
+        batched_input: List[str],
+        generation_pipeline: transformers.Pipeline,
+        generation_config: transformers.GenerationConfig,
+    ) -> List[List[str]]:
+        """
+        Answer questions with a context to the given text files contents by a pretrained LLM model in given pipeline.
+        """
+        return self._answer_questions(
+            questions_amount=questions_amount,
+            batched_input=batched_input,
+            generation_pipeline=generation_pipeline,
+            generation_config=generation_config,
+        )
+
+
+class PollQuestionHandler(QuestionHandler):
+    """
+    A class for handling questions answering for poll type questions.
+    These type of question are answered by asking the same question multiple times
+    and choosing the most common answer or the average answer.
+    """
+
+    class ConfigKeys:
+        """
+        A class for handling questions answering for poll type questions.
+        These type of question are answered by asking the same question multiple times
+        and choosing the most common answer or the average answer.
+        """
```

Comment:
don't count on co-pilot too much, this is a copy of the main class. You can simply say `"Static class to hold all the possible poll question configurations options keys"`.